week-10/10-tech-lab.adoc

```diff
@@ -0,0 +1,228 @@
+
+* something installed that doesn't need to be
+* something that should be there, that isn't
+* permissions that need to be tightened down.
+
+* 1 vulnerability fixed: 10 points
+* 2 : 15 points
+* 3 : 20 points.
+* 10 or more: 30 points
+
+In order to gain the extra credit:
+
+* Create a new VM and run the vulnerability scanner.
+* Save the output to a text file, named "initial-output.log"
+* Run the necessary commands to fix the vulnerability.
+* Put them into a shell script.
+* Run the scanner again, saving the output to "final-output.log" and showing that they are fixed
+
+IMPORTANT: Submit the 2 output files, and your script for extra credit.
+
+You can script commands like
+
+ apt-get install <some program>
+
+You can't script an interactive text editor (like nano). To change configuration files, sed is your friend. Simple example:
+
+....
+char@seis664:~$ echo myScriptContent > test.txt
+char@seis664:~$ cat test.txt
+myScriptContent
+char@seis664:~$ sed -i s/Script/New/g test.txt
+char@seis664:~$ cat test.txt
+myNewContent
+....
+
+You may need to Google "regular expressions."
+
+=== Web vulnerability testing scan
+(if time)
+
+When you have reflected enough on the Java deserialization vulnerability and the implications of that article, move on to a Web vulnerability. We will install 
+Java and apache2 on your VM. (Don't try to install these on the main server; you won't be able to.)
+
+ON YOUR VM, perform an update and install java and apache2.  You can check if apache2 got installed by issuing the curl command.
+
+....
+sudo apt-get update
+sudo apt-get install -y default-jdk apache2
+curl localhost:80
+....
+
+Install owasp-zap.  
+Go to https://www.zaproxy.org/download/ and scroll down until you find the link 'ZAP Linux Repos'.  Click on this link to go to the download page.
+On the download page, select your operating system, then click on 'Add repository and install manually'.  
+
+For example, if you chose the Ubuntu operation system:
+
+image::owasp_zap_download.png[]
+
+Run the commands that get displayed for the operating system that you chose.  After owasp-zap is installed, use the 'find' command to determine where zap.sh 
+was installed.  Run it and redirect its output to a file, e.g. zap_sh_out, so you can view it later.
+
+....
+sudo find / -name zap.sh
+sudo <zap.sh* directory>/zap.sh -cmd -quickurl http://localhost:80 > zap_sh_out
+....
+
+We run this script externally to the machine being tested, and give it the URL. It then probes the URL and the server, as a form of penetration testing. It will again generate a number of findings. Research them and figure out if there is an easy fix. Share with class on Slack.
+
+NOTE: You will get raw XML dumped to the terminal. You can cut and paste this to an *.xml document on your workstation and open it with a browser for an easier view.
+
+=== Optional ITSM process
```

Comment:
the ITSM section should reference JIra SD not ITOP.

Reply:
Fixed.